### PR TITLE
fix detection of `RegExp.prototype.hasIndices`

### PR DESCRIPTION
**Pull request by axiac**
Solves issue #1863.

### CODE REVIEW

File: data-es2016plus.js [hunk 1]
```diff
@@ -6641,6 +6641,8 @@ exports.tests = [
           var actual = [];
           var p = new Proxy({}, { get: function(o, k) { actual.push(k); return o[k]; }});
           Object.getOwnPropertyDescriptor(RegExp.prototype, 'flags').get.call(p);
+          expected.sort();
+          actual.sort();
```
Comment (ljharb) [inline data-es2016plus.js:6645]: I'm confused why this fixes anything - part of the spec requirements is that the list is already alphabetized here.

**Review by zloirock — COMMENTED** [review]
Here missed some flags like `.unicodeSets` - this test is incorrect because of this.

Old V8 (< 11.1 -> Node < 20.0) implementations had a bug with an order of getting keys in `.flags` - it's a fair reason why this test fails in old Node versions.

File: data-es2016plus.js (hunk 1 above)
Comment (axiac) [reply data-es2016plus.js:6645]: Before sorting, on Node.js 16.19.1 `expected` and `actual` look like this:
```
expected: [
  'hasIndices',
  'global',
  'ignoreCase',
  'multiline',
  'dotAll',
  'unicode',
  'sticky'
]
actual: [
  'global',
  'ignoreCase',
  'multiline',
  'dotAll',
  'unicode',
  'sticky',
  'hasIndices'
]
```
Given that the order should not matter, isn't an object a better container for these values?

Comment (ljharb) [reply data-es2016plus.js:6645]: order should matter.

Comment (axiac) [reply data-es2016plus.js:6645]: Why should order matter? The flags are independent.

Comment (ljharb) [reply data-es2016plus.js:6645]: The spec mandates the order.

Comment (axiac) [reply data-es2016plus.js:6645]: The specs also require that the flag `hasIndices` is implemented but not all JavaScript engines support it at this moment. That's why this compatibility table exists, isn't it?

Node.js 16 implements `hasIndices` despite not returning the flag names in the expected order.
In fact, `RegExp.prototype.flags` returns them in the expected order. It's only the property descriptor the one that lists them in a different order.

Comment (ljharb) [reply data-es2016plus.js:6645]: Which means node 16 isn’t compliant.

I’d be fine splitting this test into two - one that checks the functionality and existence of the accessors, and another that checks the ordering - but spec compliance is binary, and if the order is wrong, it fails.

Comment (axiac) [reply data-es2016plus.js:6645]: I don't get it. Why is the order so important?

Comment (ljharb) [reply data-es2016plus.js:6645]: Every single thing the spec mandates is equally critically important - compliance is binary, either something does, or doesn’t.

**Comment by axiac** [comment]
I thought that the goal of this project is to help the developers find out what ES functionalities are implemented by each JavaScript engine.
Apparently you do not care about the developers.
:-(

**Comment by ljharb** [comment]
@axiac it is, and the ordering of the flags is a part of that.

**Pull request closed without merging**